src/base/bittorrent/torrentinfo.cpp

```diff
@@ -246,6 +246,28 @@ QVector<int> TorrentInfo::fileIndicesForPiece(int pieceIndex) const
     return res;
 }
 
+QByteArray TorrentInfo::hashForPiece(const int pieceIndex) const
+{
+    if (!isValid() || (pieceIndex < 0) || (pieceIndex >= piecesCount()))
+        return {};
+    return { m_nativeInfo->hash_for_piece_ptr(pieceIndex), libtorrent::sha1_hash::size };
+}
+
+QVector<QByteArray> TorrentInfo::piecesHashes() const
+{
+    if (!isValid())
+        return {};
+
+    const int count = piecesCount();
+    QVector<QByteArray> hashes;
+    hashes.reserve(count);
+
+    for (int i = 0; i < count; ++i)
+        hashes.append(hashForPiece(i));
```

Comment:
Shouldn't the method be the correct behavior, independent of particular use cases? If we don't want these checks inside loop we should abolish this method and use its code in loop directly (without checks part) or as lambda. At least, we should make `hashForPiece()` private.